src/api/port.js

```diff
@@ -0,0 +1,242 @@
+	return isPortRecord(port, true);
+}
+
+/**
+ * Check if the typologie is one of the two accepted value
+ * @param {*} typology
+ * @param {bool} doThrow
+ */
+export function isTypology(typology, doThrow) {
+	if (typology === PORT_SOURCE || typology === PORT_SINK) {
+		return true;
+	}
+	if (doThrow) {
+		throw new Error(`Should be a typology 'SOURCE' or 'SINK' was given ${typology.toString()}`);
+	}
+	return false;
+}
+
+/**
+ * @param {PortRecord} port
+ * @returns {string}
+ */
+export function getId(port) {
+	const whatver = isPortRecordElseThrow(port);
+	if (whatver) {
+		return port.get('id');
+	}
+	return false;
+}
+
+/**
+ * @param {string}
+ * @param {PortRecord}
+ * @returns {PortRecord}
+ */
+const setId = curry((id, port) => {
+	if (isPortRecord(port) && typeof id === 'string') {
+		return port.set('id', id);
+	}
+	throw new Error(`id should be a string was given ${id.toString()}`);
+});
+
+/**
+ * @param {PortRecord} port
+ * @returns {string}
+ */
+export function getNodeId(port) {
+	if (isPortRecord(port, true)) {
+		return port.get('nodeId');
+	}
+	return false;
+}
+
+/**
+ * @param {string} nodeId
+ * @param {PortRecord} port
+ * @returns {PortRecord}
+ */
+export const setNodeId = curry((nodeId, port) => {
+	if (isPortRecord(port, true) && typeof nodeId === 'string') {
+		return port.set('nodeId', nodeId);
+	}
+	throw new Error(`nodeId should be a string was given ${nodeId.toString()}`);
+});
+
+/**
+ * @param {PortRecord} port
+ * @returns {PositionRecord}
+ */
+export function getPosition(port) {
+	if (isPortRecord(port, true)) {
+		return port.getIn(positionSelector);
+	}
+	return false;
+}
+
+/**
+ * @param {PositionRecord} position
+ * @param {PortRecord} port
+ * @returns {Port}
+ */
+export const setPosition = curry((position, port) => {
+	if (isPortRecord(port, true) && isPositionRecord(position, true)) {
+		return port.setIn(positionSelector, position);
+	}
+	return false;
+});
+
+/**
+ * @param {PortRecord} port
+ * @returns {string}
+ */
+export function getComponentType(port) {
+	if (isPortRecord(port, true)) {
+		return port.getIn(componentTypeSelector);
+	}
+	return false;
+}
+
+/**
+ * @param {string} componentType
+ * @param {PortRecord} port
+ * @returns {PortRecord}
+ */
+export const setComponentType = curry((componentType, port) => {
+	if (isPortRecord(port, true) && typeof componentType === 'string') {
+		return port.setIn(componentTypeSelector, componentType);
+	}
+	throw new Error(`componentType should be a string was given ${componentType.toString()}`);
+});
+
+/**
+ * @param {PortRecord} port
+ * @returns {String}
+ */
+export function getTypology(port) {
+	if (isPortRecord(port, true)) {
+		return port.getIn(portTopologySelector);
+	}
+	return false;
+}
+
+/**
+ * @param {string} typology
+ * @param {PortRecord} port
+ * @returns {PortRecord}
+ */
+export const setTypology = curry((typology, port) => {
+	if (isPortRecord(port, true && isTypology(typology))) {
```

Comment:
I guess you can remove that `true` statement.

Reply:
wrongly placed parenthèsis